aleph/logic/xref.py

```diff
@@ -272,12 +272,38 @@ def xref_collection(collection):
     )
     log.info(f"[{collection}] Clearing previous xref state....")
     delete_xref(collection, sync=True)
-    delete_entities(collection.id, origin=ORIGIN, sync=True)
+    delete_entities(collection.id, origin='xref', sync=True)
+
+    batch = []
+    batch_size = int(SETTINGS.XREF_SCROLL_SIZE)  # Ensure batch_size is an integer
+    for entity in iter_entities(collection_id=collection.id):
+        batch.append(entity['id'])
+        if len(batch) >= batch_size:
+            enqueue_xref_batch(collection, batch)
+            batch = []
+
+    if batch:
+        enqueue_xref_batch(collection, batch)
+
+    log.info(f"[{collection}] Xref batches enqueued, processing will continue in the background.")
+    refresh_collection(collection.id)
```

Comment:
I'll have to check a couple of things as well before I could speak on that, so I will get back on this point, I think by tomorrow.

Reply:
Had another look: The cached data is mostly statistics about the collection (for example, number of entities by schema, …).

Computing these stats uses ES aggregation queries that can be quite expensive, but in this case they are always limited to a single collection, so they should usually be relatively quick to compute. Additionally, `refresh_collection` doesn’t actually recompute the stats, it invalidates them so that they are recomputed the next time they are accessed.

I think we should be fine calling `refresh_collection` after each batch of entities.